src/pool/pool_jemalloc.c

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright (C) 2022-2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+
+#include <jemalloc/jemalloc.h>
+
+#include <umf/memory_pool.h>
+#include <umf/memory_pool_ops.h>
+#include <umf/pools/pool_jemalloc.h>
+
+#define MALLOCX_ARENA_MAX (MALLCTL_ARENAS_ALL - 1)
+
+#ifdef _WIN32
+#define __TLS __declspec(thread)
+#else
+#define __TLS __thread
+#endif
+
+static __TLS enum umf_result_t TLS_last_allocation_error;
+
+typedef struct jemalloc_memory_pool_s {
+    unsigned int arena_zero; // index of the first jemalloc arena of this pool
+    unsigned int arena_map_len; // is power of 2
+
+    umf_memory_provider_handle_t hProvider;
+} jemalloc_memory_pool_t;
+
+static jemalloc_memory_pool_t *arena_registry_g[MALLCTL_ARENAS_ALL];
+
+static jemalloc_memory_pool_t *get_kind_by_arena(unsigned arena_ind) {
+    // there is no way to obtain MALLOCX_ARENA_MAX from jemalloc
+    // so this checks if arena_ind does not exceed assumed range
+    assert(arena_ind < MALLOCX_ARENA_MAX);
+
+    return arena_registry_g[arena_ind];
+}
+
+static void *arena_extent_alloc(extent_hooks_t *extent_hooks, void *new_addr,
+                                size_t size, size_t alignment, bool *zero,
+                                bool *commit, unsigned arena_ind) {
+    (void)extent_hooks; // unused
+    (void)new_addr;     // unused
+
+    jemalloc_memory_pool_t *pool = get_kind_by_arena(arena_ind);
```

Comment:
I don't understand - so we have several pools? A separate pool for each arena?

Reply:
Changed

Reply:
As regards  memory tracking capabilities I am working on it.